vector/src/main/java/im/vector/app/features/crypto/verification/VerificationBottomSheet.kt

```diff
@@ -135,6 +133,14 @@ class VerificationBottomSheet : VectorBaseBottomSheetDialogFragment<BottomSheetV
         }
     }
 
+    override fun onCancel(dialog: DialogInterface) {
+        viewModel.queryCancel()
+        withState(viewModel) { state ->
+            showCancelVerificationDialog(state)
```

Comment:
I tried so at the beginning but I found problems : 
 - Once on Cancel is called the base bottom sheet get canceled also which kill the view model / shared view model state. at that level I do not have access to the state updated with the value of userWantsToCancel, despite of when clicking on back button, the state keeps alive. 
 - I'm sure using invalidate or view states are the best way to handle, but I can't get the data needed. Welcome to any other solution.

Reply:
Okay I see. The only solution I have in mind is to not call `super.onCancel()` so that the `BottomSheet` is not canceled and destroyed. Meaning we will have dialog showing above to confirm the cancellation. Then we are able to call `dismiss()` when receiving the corresponding ViewEvent like it was done previously. But this is different UX. @bmarty What do you think?  
By the way, there are `confirmCancel()` and `continueFromCancel()` which are no more used in the `VerificationBottomSheetViewModel` due to the changes

Reply:
I don't feel too strongly on this, if we want to have the UX of closing the sheet, showing a dialog and maybe relaunching the sheet, showing the dialog in the cancel seems like the simplest way

modelling the dialog as state could be nice! but I'm not sure we have any examples of this in the app :thinking: 

perhaps capturing the `onCancel` and mapping to a `ViewEvent` could be an intermediate step?

```kotlin
override fun onCancel() {
  // skip super
  viewEvents.tryEmit(Cancel)
}

... 

onEvent {
  when(event) {
    Cancel -> withState(viewModel) { state ->
      showCancelVerificationDialog(state)
      super.onCancel()
    }
}
```

Reply:
Let's keep the current implementation then. But @ahmed-radhouane , can you remove the no more used methods `confirmCancel()` and `continueFromCancel()` from the `VerificationBottomSheetViewModel`?